### ISSUE TITLE
Unable to connect to peers in Bodies stage

### ISSUE THREAD

**Issue by 0xpantera** [paradigmxyz/reth#3647]
### Describe the bug

I'm running a reth/lighthouse validator node on goerli. After reaching 98.2% of the `Bodies` stage reth has been unable to connect to peers for ~12 hours. It stays at 0 peers connected and whenever it establishes a session with a peer it's immediately followed by a disconnect command for the session. I have tried restarting reth.

### Steps to reproduce

I'm running the reth/lighthouse on a spare laptop with the following specs:
- 32GB Dual Channel DDR4 at 2666MHz (2x16GB)
- 1 TB M.2 SSD
- 2 TB 2.5" Solid State Hybrid Drive
- 4.1 GHz i7-875OH (2.2 up to 4.1 GHz - 9MB Cache - 6 cores - 12 threads)

Reth, lighthouse-beacon and lighthouse-validator are running as services in `/etc/systemd/system/` with the following configuration (*** for possibly sensitive fields):
Reth:
```
[Unit]
Description=Reth Execution Client (Goerli Test Network)
After=network.target
Wants=network.target

[Service]
User=reth
Group=reth
Type=simple
Restart=always
RestartSec=5
Environment="RUST_LOG=info"
ExecStart=/usr/local/bin/reth node \
  --datadir /var/lib/reth \
  --authrpc.jwtsecret=***/jwt.hex \
  --authrpc.addr 127.0.0.1 \
  --authrpc.port 8551 \
  --http --http.api "admin,debug,eth,net,trace,txpool,web3,rpc" \
  --chain goerli \
  --metrics 127.0.0.1:9001

[Install]
WantedBy=default.target
```

Lighthouse beacon
```
[Unit]
Description=Lighthouse Consensus Client BN (Goerli Test Network)
Wants=network-online.target
After=network-online.target

[Service]
User=lighthousebeacon
Group=lighthousebeacon
Type=simple
Restart=always
RestartSec=5
ExecStart=/usr/local/bin/lighthouse bn \
  --network goerli \
  --datadir /var/lib/lighthouse \
  --http \
  --execution-endpoint http://localhost:8551 \
  --execution-jwt ***/jwt.hex \
  --checkpoint-sync-url https://sync-goerli.beaconcha.in \
  --metrics \
  --purge-db

[Install]
WantedBy=multi-user.target
```

Lighthouse validator
```
[Unit]
Description=Lighthouse Consensus Client VC (Goerli Test Network)
Wants=network-online.target
After=network-online.target

[Service]
User=lighthousevalidator
Group=lighthousevalidator
Type=simple
Restart=always
RestartSec=5
ExecStart=/usr/local/bin/lighthouse vc \
  --network goerli \
  --datadir /var/lib/lighthouse \
  --suggested-fee-recipient *** \
  --graffiti ***

[Install]
WantedBy=multi-user.target
```

### Node logs

```text
Jul 06 14:38:15 pop-os reth[23682]: 2023-07-06T20:38:15.562190Z  INFO reth::node::events: Forkchoice updated head_block_hash=0x4facb1b909211629059eaf11679096a90f796c91ecf0d2388ef05530a2411467 safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:38:17 pop-os reth[23682]: 2023-07-06T20:38:17.360695Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:38:23 pop-os reth[23682]: 2023-07-06T20:38:23.282389Z  INFO net: Session established remote_addr=65.109.66.166:20303 client_version=erigon/v2.40.1-stable-94b4cee5/linux-amd64/go1.18.1 peer_id=0x57cd06afc853a6c9a11c0b6d576105b496ffe4f2b7a5a605d5eae2f77f26e42edd55906aa89dc7bab6c6e742c7c19d235f817f714e99808eaa0cbf86c386c0e9 total_active=1
Jul 06 14:38:25 pop-os reth[23682]: 2023-07-06T20:38:25.447449Z  INFO reth::node::events: Forkchoice updated head_block_hash=0x5765c1f29c3c065682b4ddcdaf05d8cee6422a2b72257d9e328c515da46de0af safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:38:34 pop-os reth[23682]: 2023-07-06T20:38:34.164544Z  INFO net: Session established remote_addr=65.21.196.48:30304 client_version=Geth/v1.11.6-stable-ea9e62ca/linux-amd64/go1.20.3 peer_id=0x7f47bbd5e7cb2516d027bec110b837c423bfab3772741287fa8ee54a638fb1e67630865ecb44b060223327432bdd71eda2415dad34651f07fec245e8c61fad5d total_active=1
Jul 06 14:38:37 pop-os reth[23682]: 2023-07-06T20:38:37.903726Z  INFO reth::node::events: Forkchoice updated head_block_hash=0x369aec49f4bc156bacd825e5680d554103d4697107e1652057bb047f3226acb9 safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:38:47 pop-os reth[23682]: 2023-07-06T20:38:47.360791Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:38:48 pop-os reth[23682]: 2023-07-06T20:38:48.963981Z  INFO reth::node::events: Forkchoice updated head_block_hash=0xa58d75350176143134970ae39784b5e83458b38561e5e90456fb73350da19e6a safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:39:17 pop-os reth[23682]: 2023-07-06T20:39:17.361301Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:39:18 pop-os reth[23682]: 2023-07-06T20:39:18.823117Z  INFO net: Session established remote_addr=65.21.196.48:30304 client_version=Geth/v1.11.6-stable-ea9e62ca/linux-amd64/go1.20.3 peer_id=0x7f47bbd5e7cb2516d027bec110b837c423bfab3772741287fa8ee54a638fb1e67630865ecb44b060223327432bdd71eda2415dad34651f07fec245e8c61fad5d total_active=1
Jul 06 14:39:18 pop-os reth[23682]: 2023-07-06T20:39:18.891256Z  INFO net: Session established remote_addr=100.27.34.227:30303 client_version=Nethermind/v1.19.2+42ba08b7/linux-x64/dotnet7.0.5 peer_id=0xbbe8ea836cc682a582811894a692dc526234535d1a902e503a6f34f6ca9c3ca4f8b7aaee46de4eece51ea5bc2649be014f24cf32d5781e612931faa801e9d04d total_active=1
Jul 06 14:39:19 pop-os reth[23682]: 2023-07-06T20:39:19.081241Z  INFO net: Session established remote_addr=91.199.118.245:40303 client_version=Nethermind/v1.17.2+4faa7314/linux-x64/dotnet7.0.4 peer_id=0x001cfb117d33c83c4cd137690563eca9609da02e33b9fe41feb79995a65df26063755eac848f4bca01750c1b13be435ab730e102145351a5fa1b171ba86d5702 total_active=2
Jul 06 14:39:19 pop-os reth[23682]: 2023-07-06T20:39:19.356246Z  INFO net: Session established remote_addr=54.248.47.234:30303 client_version=Geth/v1.11.6-stable/linux-amd64/go1.20.3 peer_id=0x6fbc0c561a412cbc2e2195936b165e9713386a712c5188b96ba03b067de0adc8047ff2927123ef9718dd163eb07d287db352c9b771588fe611daee207767666b total_active=2
Jul 06 14:39:22 pop-os reth[23682]: 2023-07-06T20:39:22.546349Z  INFO net::session: Received disconnect command for session reason=Some(DisconnectRequested) remote_peer_id=0x001cfb117d33c83c4cd137690563eca9609da02e33b9fe41feb79995a65df26063755eac848f4bca01750c1b13be435ab730e102145351a5fa1b171ba86d5702
Jul 06 14:39:27 pop-os reth[23682]: 2023-07-06T20:39:27.709133Z  INFO reth::node::events: Forkchoice updated head_block_hash=0xaf567cf21651eceb53bf61055eb0baf805d801c772345e6532489a8ab80416a0 safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:39:30 pop-os reth[23682]: 2023-07-06T20:39:30.666734Z  INFO net::session: Received disconnect command for session reason=Some(DisconnectRequested) remote_peer_id=0x6fbc0c561a412cbc2e2195936b165e9713386a712c5188b96ba03b067de0adc8047ff2927123ef9718dd163eb07d287db352c9b771588fe611daee207767666b
Jul 06 14:39:47 pop-os reth[23682]: 2023-07-06T20:39:47.360866Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:39:49 pop-os reth[23682]: 2023-07-06T20:39:49.795273Z  INFO reth::node::events: Forkchoice updated head_block_hash=0xf90aa579a46f8683a59395e2909b94f868194cd75cfb75251ceb1050d085f10b safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
Jul 06 14:40:03 pop-os reth[23682]: 2023-07-06T20:40:03.911608Z  INFO net: Session established remote_addr=65.21.196.48:30304 client_version=Geth/v1.11.6-stable-ea9e62ca/linux-amd64/go1.20.3 peer_id=0x7f47bbd5e7cb2516d027bec110b837c423bfab3772741287fa8ee54a638fb1e67630865ecb44b060223327432bdd71eda2415dad34651f07fec245e8c61fad5d total_active=1
Jul 06 14:40:17 pop-os reth[23682]: 2023-07-06T20:40:17.360588Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:40:21 pop-os reth[23682]: 2023-07-06T20:40:21.636743Z  INFO net: Session established remote_addr=3.94.98.214:30303 client_version=Geth/v1.11.6-stable/linux-amd64/go1.20.3 peer_id=0xe050cc5f3dfa32cbdb3c10fdd5fa071a5910f6f07f53d319ea7c4f6739c962f86b060806f0610df454beccfb85cd9440b0131845d5a65852b18e5a30e534e3e1 total_active=1
Jul 06 14:40:38 pop-os reth[23682]: 2023-07-06T20:40:38.395298Z  INFO net::session: Received disconnect command for session reason=Some(DisconnectRequested) remote_peer_id=0xe050cc5f3dfa32cbdb3c10fdd5fa071a5910f6f07f53d319ea7c4f6739c962f86b060806f0610df454beccfb85cd9440b0131845d5a65852b18e5a30e534e3e1
Jul 06 14:40:47 pop-os reth[23682]: 2023-07-06T20:40:47.360526Z  INFO reth::cli: Status connected_peers=0 stage=Bodies checkpoint=98.2% eta=unknown
Jul 06 14:40:48 pop-os reth[23682]: 2023-07-06T20:40:48.938402Z  INFO net: Session established remote_addr=65.21.196.48:30304 client_version=Geth/v1.11.6-stable-ea9e62ca/linux-amd64/go1.20.3 peer_id=0x7f47bbd5e7cb2516d027bec110b837c423bfab3772741287fa8ee54a638fb1e67630865ecb44b060223327432bdd71eda2415dad34651f07fec245e8c61fad5d total_active=1
Jul 06 14:40:51 pop-os reth[23682]: 2023-07-06T20:40:51.420309Z  INFO reth::node::events: Forkchoice updated head_block_hash=0xc9a66879ff259d82babee1491a7469934118edacb5edbb045e0e7d8172f4ca1a safe_block_hash=0x61a5bc045be226d986cb4db401c9031559e9e43d5afbe698d3af9e83fc613576 finalized_block_hash=0x2e1d00d5c555d4ead1f8d0635c5a1ee54035b5affe079deca5477754f861d052 status=Syncing
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth Version: 0.1.0-alpha.1

### What database version are you on?

Current database version: 1
Local database version: 1

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by 0xpantera** [comment]
Similar thing happened at different stage. I have been at 98% for about 3 days now without being able to connect to peers:

**Comment by mattsse** [comment]
sorry to hear that,
there have been reports that peering is especially bad on goerli for some reason

could you try sepolia as an alternative? should only take a few hours

**Comment by onbjerg** [comment]
Syncing testnets can be very very hard because of the low quality peerset. I have been unable to peer on Goerli myself, but getting peers on Sepolia without issue.

For more context, would it be possible if you could screenshot the graphs of the "Disconnect reason" pie on the Grafana dashboard? (filter out too many peers as it likely takes up 99% of the pie)

**Comment by 0xpantera** [comment]
Thanks for your responses. Will probably move to Sepolia then. I'm attaching the Grafana "Disconnect reason" panel with "too many peers" filtered out. 
- DisconnectRequested: 2519
- SubprotocolSpecific: 1594
- ClientQuitting: 402
- AlreadyConnected: 309
- UselessPeer: 13
![grafana-reth-p2p-panel](https://github.com/paradigmxyz/reth/assets/6171312/e5ff173f-349a-4e23-a14c-603e145d45ec)

**Comment by onbjerg** [comment]
Interesting, I don't really see `DisconnectRequested` that often on other networks, @mattsse do you off-hand know any reason peers would disconnect with this reason? My only guess would be that peers would want to disconnect from us with this reason if 1) they only want synced nodes, 2) they find us to be low quality for some reason? i.e. poor latency, unsure if any nodes filter on this rn or 3) we have a low reputation for some reason

**Comment by 0xc926ddba** [comment]
I have a similar issue, though my grafana pie-graph only is labeled $x-series. This is the stock dashboard, so not sure how mine has different labels. As you can see, once I hit sync finished after IndexAccountHistory step was completed, all peers disconnect and then I can't seem to establish connections that the other side doesn't drop anymore. More info about my setup here: https://github.com/paradigmxyz/reth/discussions/3369

Link to grafana dashboard snapshot: https://snapshots.raintank.io/dashboard/snapshot/C80oDckicKcftBzSE4D3NNkjE13nP6T9?orgId=2&refresh=30s

**Comment by 0xc926ddba** [comment]
Nevermind, this is fixed after using latest master instead of 0.1.0-alpha.3. Probably due to recent bugfix https://github.com/paradigmxyz/reth/pull/3783

**Comment by gakonst** [comment]
Yep exactly! Sorry about that folks. Closed by #3783.
